### PR TITLE
Marketplace cart page

### LINKED ISSUE #1679: Cart page
## UI
[Invision](https://projects.invisionapp.com/d/main#/console/19020838/403973310/preview)

## Acceptance criteria
- Display the distribution deals from the cart film by film
- Same table and data to display as in dashboard - deal page - but only Territory, Excluded Territory, Start Date, End Date & Medias
- Add edit and delete button on each raw of the distribution deal tables
- Consult your Current Offers and Deals button redirects to /deals
- Add Exploitation Rights button redirects to the movie page, tab avails
- Continue browsing the Library redirects to /search
- @phpgeekfr need to add titleDetails object

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Create a marketplace state to store the cart locally.
Cart isn't linked to an organization (not in the database), but it's in the localstorage.
fix #1679

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/+state/marketplace.query.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(protected store: MarketplaceStore, private movieQuery: MovieQuery) {
+    super(store);
+  }
+
+  /** The list of titles in the wishlist */
+  wishlist$: Observable<Movie[]> = this.select('wishlist').pipe(
+    switchMap(movieIds => this.movieQuery.selectMany(movieIds))
+  )
+
+  /** Check if a titl is in the wishlist */
```
Comment (ioaNikas) [inline apps/catalog/catalog/src/app/marketplace/+state/marketplace.query.ts]: title

File: apps/catalog/catalog/src/app/marketplace/components/wishlist-button/wishlist-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }
+
+  
+  public addToWishlist() {
+    const movie = this.movieQuery.getEntity(this.movieId);
+    const title = movie.main.title.international;
+    this.cartService.updateWishlist(movie);
+    this.analytics.event(AnalyticsEvents.addedToWishlist, {
+      movieId: movie.id,
+      movieTitle: movie.main.title.original
```
Comment (ioaNikas) [inline apps/catalog/catalog/src/app/marketplace/components/wishlist-button/wishlist-button.component.ts]: Why use title.original for analytics ?

File: libs/movie/src/lib/movie/components/banner/banner.component.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ export class BannerComponent {
   public date: number;
 
   @Input() set movie(movie: Movie) {
+    console.log(movie);
```
Comment (ioaNikas) [inline libs/movie/src/lib/movie/components/banner/banner.component.ts]: Forgot to remove

File: apps/catalog/catalog/src/app/marketplace/components/wishlist-button/wishlist-button.component.ts (hunk 2 above)
Comment (GrandSchtroumpf) [reply apps/catalog/catalog/src/app/marketplace/components/wishlist-button/wishlist-button.component.ts]: I copy pasted the code that was already there

**Pull request merged**